log4j-1.2-api/src/test/java/org/apache/log4j/CategoryTest.java

```diff
@@ -334,25 +335,43 @@ private <M extends Message> void testMessageImplementation(
     public void testAddAppender() {
         try {
             final Logger rootLogger = LogManager.getRootLogger();
-            int count = version1Appender.getEvents().size();
+            version1Appender.clear();
             rootLogger.addAppender(version1Appender);
             final Logger logger = LogManager.getLogger(CategoryTest.class);
             final org.apache.log4j.ListAppender appender = new org.apache.log4j.ListAppender();
             appender.setName("appender2");
             logger.addAppender(appender);
             // Root logger
             rootLogger.info("testAddLogger");
-            assertEquals(++count, version1Appender.getEvents().size(), "adding at root works");
-            assertEquals(0, appender.getEvents().size(), "adding at child works");
+            assertThat(version1Appender.getEvents())
+                    .as("check root logger events")
+                    .extracting(LoggingEvent::getRenderedMessage)
+                    .containsExactly("testAddLogger");
+            assertThat(appender.getEvents())
+                    .as("check logger '%s' events", logger.getName())
+                    .isEmpty();
```

Comment:
These improvements in `CategoryTest` are the reason for the changes in `LogEventWrapper/LogEventAdapter`.
I could push these in a separate PR.